ex01.exs

```diff
@@ -26,8 +26,38 @@ defmodule Ex01 do
         2 is the program well laid out,  appropriately using indentation,
           blank lines, vertical alignment
   """
+
+  #######
+  # API #
+  #######
+  
+  def new_counter(value \\ 0) do
+    spawn Ex01, :counter, [ value ]
+  end
+
```

Comment:
I like the fact that you've split out the implementation into functions.

### +2

I'm not sure about the `get_response` function. If it was written

~~~ elixir
  defp get_response do
    receive do
      { :next_value, value} -> value
    end
  end
~~~

then it would be slightly more robust (because it would wait for an explicit `:next_value` and it would make `next_value` simpler

~~~ elixir
 def next_value(counter_pid) do
    send counter_pid, { :next, self() }
    get_response()
  end
~~~